File: Chapter03/NamedFontSizes/NamedFontSizes/NamedFontSizes/NamedFontSizesPage.cs
using System;
using System.Linq;
using Xamarin.Forms;

namespace NamedFontSizes
{
    public class NamedFontSizesPage : ContentPage
    {
        public NamedFontSizesPage()
        {
            FormattedString formattedString = new FormattedString();
            NamedSize[] namedSizes = 
            {
                NamedSize.Default, NamedSize.Micro, NamedSize.Small,
                NamedSize.Medium, NamedSize.Large
            };

            foreach (NamedSize namedSize in namedSizes)
            {
                double fontSize = Device.GetNamedSize(namedSize, typeof(Label));

                formattedString.Spans.Add(new Span
                    {
                        Text = String.Format("Named Size = {0} ({1:F2})",
                                             namedSize, fontSize),
                        FontSize = fontSize
                    });

                if (namedSize != namedSizes.Last())
                {
                    formattedString.Spans.Add(new Span
                        {
                            Text = Environment.NewLine + Environment.NewLine
                        });
                }
            }

            Content = new Label
            {
                FormattedText = formattedString,
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center
            };
        }
    }
}

File: Chapter03/FS/NamedFontSizes/NamedFontSizes/NamedFontSizes/NamedFontSizesPage.fs
namespace NamedFontSizes

open System
open Xamarin.Forms

type NamedFontSizesPage() = 
    inherit ContentPage()
    let formattedString = FormattedString()
    let namedSizes = 
        [| 
            NamedSize.Default
            NamedSize.Micro
            NamedSize.Small
            NamedSize.Medium
            NamedSize.Large 
        |]

    do for namedSize in namedSizes do
        let fontSize = Device.GetNamedSize(namedSize, typeof<Label>)
        do formattedString.Spans.Add(Span(Text = String.Format("Named Size = {0} ({1:F2})",
                                                               namedSize,
                                                               fontSize),
                                          FontSize = fontSize))
        if namedSize <> Array.last(namedSizes)
            then formattedString.Spans.Add(Span(Text = Environment.NewLine + Environment.NewLine))

    do base.Content <- Label(FormattedText = formattedString,
                             VerticalOptions = LayoutOptions.Center,
                             HorizontalOptions = LayoutOptions.Center)


    